.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,25 @@
+name: Build Project
+
+on:
+  pull_request:
```

Comment:
Is this really the "right" and common way in public repositories?
I guess anyone is able to run our pipelines now, without any approval or something.

Reply:
Firstly, CI pipelines are run at the branch were they are defined. [Just running those from the main branch](https://github.com/orgs/community/discussions/53430) is not supported at the moment.

There would be a [Push ruleset](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets#push-rulesets), but these are not available for public repos:
> Push rulesets are available for the GitHub Team plan in internal and private repositories, and forks of repositories that have push rulesets enabled.

I've set the action settings to the following:

* **Action Permissions:** Allow aditosoftware actions and reusable workflows. Any action or reusable workflow defined in a repository within aditosoftware can be used.
* Fork pull request workflows from outside collaborators: Require approval for all outside collaborators

This should put us in control of running the workflows when changes from the forks are coming.



----


For running workflows in pull requests, I looked at the  [example node.js workflow](https://github.com/aditosoftware/eslint-config-adito-platform/new/main?filename=.github%2Fworkflows%2Fnode.js.yml&workflow_template=ci%2Fnode.js)